src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,46 @@
+import classNames from 'classnames';
+import React from 'react';
+import * as RadixDialog from '@radix-ui/react-dialog';
+import { CloseIcon } from '../icons';
+
+import styles from './dialog.module.scss';
+
+export interface DropdownMenuProps extends React.PropsWithChildren {
+    title: string;
+    description?: string;
+    open: boolean;
+    onOpenChange: (open: boolean) => void;
+    trigger: React.ReactElement;
+}
+
+export const Dialog = ({
+    trigger,
+    title,
+    description,
+    open,
+    onOpenChange,
+    children,
+}: DropdownMenuProps) => (
+    <RadixDialog.Root open={open} onOpenChange={onOpenChange}>
+        <RadixDialog.Trigger asChild>{trigger}</RadixDialog.Trigger>
+        <RadixDialog.Portal>
+            <RadixDialog.Overlay className={styles.overlay} />
+            <RadixDialog.Content className={styles.content}>
+                <RadixDialog.Title className={styles.title}>{title}</RadixDialog.Title>
```

Comment:
you can do like this:
https://github.com/codux-templates/e-commerce-reclaim/blob/development/src/components/dropdown-menu/dropdown-menu.tsx#L33-L42

```
export const DialogTitle ...

export const DialogDescription ...

export const DialogFooter ...
```

this will give more control over dialog